utils/client.go

```diff
@@ -51,16 +93,38 @@ func (c *Client) Close() {
 
 // Read startes a blocking reader routine
 func (c *Client) Read(cb DataCallback) {
-	defer c.Close()
-	for {
-		data := &model.Message{}
-		err := c.socket.ReadJSON(data)
-		if err != nil {
-			return
+	if c.protocol == "WEBSOCKET" {
+		defer c.Close()
+		for {
+			data := &model.Message{}
+			err := c.socket.ReadJSON(data)
+			if err != nil {
+				return
+			}
+
+			cb(data)
 		}
+	} else if c.protocol == "GRPC" {
+		defer c.Close()
+		for {
+			in, err := c.stream.Recv()
+			if err != nil {
+				return
+			}
+			var data map[string]interface{}
+			data["Token"] = in.Token
+			data["DBType"] = in.DbType
+			data["Project"] = in.Project
+			data["Group"] = in.Group
+			data["Type"] = in.Type
+			data["ID"] = in.Id
+			data["Where"] = in.Where
```

Comment:
i guess `in.Where` is a []byte. The `data["where"]` must be a map[string]interface{} i guess. You should Unmarshal it first.